### ISSUE TITLE
how to train

### ISSUE THREAD

**Issue by RenXiaoman** [rl1024/SoftFormer#1]
Hi, dear author, can you provide code for training stage?Cause I do not know the procedure from pre-process to input.I would appreciate it if can provide code.   :)

**Comment by rl1024** [comment]
Hi Xiaoman, the procedure from pre-process to input differs from training the network. I have been busy recently; here is some advice to write a dataloader for preprocessing and the script for network training. If the advice is not clear enough, I will try to update a notebook around New Year's Day.

1. For the pre-processing of the image and label to the network input, given a network input with shape [C, 32, 32], suppose that you have the image with shape [C, H+32, W+32], where C is the channel number, and H and W mean the heights and widths of the images and label with shape [1, H, W]. (1) sampling training, validation, and testing samples from labels, and then output them into three CSV files; you can use **pandas** and **numpy** to achieve this. In each CSV file, we have columns like [_row,_ _col_, _label_]. Once training, you can write a data loader with PyTorch, where the label is read from the CSV file, and once you have the _row_, _col_, you can retrieve it from the image with shape[C, H+32, W+32].
2. For the training scripts, It might be useful to refer to some common network training codes; this section only deals with data reading, model definition, and model training.
